### ISSUE TITLE
Caching is broken with rxjs 5.5.0

### ISSUE THREAD

**Issue by j2L4e** [feathersjs-ecosystem/feathers-reactive#77]
With rxjs <= 5.4.3 cached observables are cleaned up when there are no subscribers left as it uses `shareReplay`. 
Apparently this isn't the desired behavior of `shareReplay`, thus it was changed in this commit: https://github.com/ReactiveX/rxjs/commit/accbcd0c5f9fd5976be3f491d454c4a61f699c4b

We need to either use the old shareReplay operator:

```
function _oldStyleShareReplay (bufferSize, windowTime, scheduler) {
  let subject;

  const connectable = multicast(function shareReplaySubjectFactory () {
    if (this._isComplete) {
      return subject;
    } else {
      return (subject = new ReplaySubject(bufferSize, windowTime, scheduler));
    }
  });
  return (source) => refCount()(connectable(source));
}
```

or generally address how caching and cleanup are done and use a different strategy to accomplish sane observable reuse.

**Comment by j2L4e** [comment]
a quick fix would be to pin the repo to `"rxjs": "5.4.x"` instead of `^5.4.0`

**Comment by daffl** [comment]
Locked it to 5.4.x in #79 for now.

**Comment by jskrzypek** [comment]
I'm curious, what is the rationale behind using a ReplaySubject via `shareReplay(1)` vs. a  BehaviorSubject via simply `share()`? Is there actually any replaying of the history going on? As far as I can tell, you'd only ever want the latest values (since that would represent the state of the server).

**Comment by j2L4e** [comment]
`share()` doesn't use a `BehaviorSubject` (which works like `ReplaySubject(1)` with an initial value) but a plain `Subject` so it just doesn't have the functionality you need for caching.

**Comment by jskrzypek** [comment]
Ah, got it... so the ideal operator would hypothetically be `shareBehavior()` (though that doesn't exist)?

**Comment by j2L4e** [comment]
The issue here is more about the `share` than the `Replay`. Generally `BehaviorSubject('blah').asObservable()` should behave exactly like `ShareReplay(1).asObservable().startWith('blah')`. As we need to get the initial value asynchronously (i.e. from the server) this is not what we need.

**Comment by jskrzypek** [comment]
I see... So the benefit to the replay is that you don't need to emit an initial null state? That makes more sense since you won't know actually know whether to return a pagination object with `data` or just an array until you see the response from the server.

**Comment by daffl** [comment]
Closed via #95
